roles/edpm_nova/scripts/derive_pci_passthrough_whitelist.py

```diff
@@ -0,0 +1,585 @@
+#!/usr/bin/python3
+
+#
+# Copyright 2019 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import json
+import os
+import re
+import yaml
+
+_PASSTHROUGH_WHITELIST_KEY = 'device_spec'
+_PCI_DEVICES_PATH = '/sys/bus/pci/devices'
+_SYS_CLASS_NET_PATH = '/sys/class/net'
+_DERIVED_PCI_WHITELIST_FILE = '/var/lib/openstack/pci_passthrough_whitelist.json'
```

Comment:
@vcandapp Need to update conf file format and path.

Reply:
@Jaganathancse, this file is retained temporarily (so that users can run the derive script and refer the whitelist like osp17). Intend to remove it later